### PR TITLE
Require empty string ("") for empty second account on `account_lines`:

### PR DESCRIPTION
**Pull request by seelabs**
A recent commit (0d0227e744a81746714dee71c4c2c63d41e3e691) broke the
parser for the rpc command `account_lines` such that an empty string was
not longer required for the second parameter. This commit fixes that bug.

### CODE REVIEW

**Comment by seelabs** [comment]
Adding some context to this PR: 

There was a bug in the RPC parser for `account_lines` such that the peers parameter was always ignored. 0d0227e744a81746714dee71c4c2c63d41e3e691 both fixed this bug and re-factored the code.  While doing the re-factoring, I mis-interpreted the `account_lines` spec and didn't release an empty string was _required_ for the peer when specifying a ledger (since it was previously always ignored). 

This commit keeps the refactoring and original bug fix and additionally requires an empty string when specifying a ledger.

**Comment by ximinez** [comment]
Good change, and it works for my use case. :+1:

File: src/ripple/net/impl/RPCCall.cpp [hunk 1]
```diff
@@ -660,25 +660,30 @@ class RPCParser
...
-            if (i > 0 && i == nParams - 1)
+            // Parameters 0 and 1 are accounts
+            if (i < 2)
+            {
+                if (parseBase58<PublicKey> (
+                        TokenType::TOKEN_ACCOUNT_PUBLIC, strParam) ||
+                    parseBase58<AccountID> (strParam) ||
+                    parseGenericSeed (strParam))
+                {
+                    jvRequest[accFields[i]] = strParam;
```
Comment (miguelportilla) [inline src/ripple/net/impl/RPCCall.cpp]: std::move(strParam) ?

Comment (miguelportilla) [inline src/ripple/net/impl/RPCCall.cpp]: Super Nitpick: Have `return parseAccountRaw2 (jvParams, "peer");` to use jss::peer

**Comment by miguelportilla** [comment]
LGTM :+1: 

I just want to note that the empty peer field string requirement only applies to the rippled command line. It can be omitted with a ledger via HTTP or WebSocket.

**Comment by seelabs** [comment]
@miguelportilla Thanks. I pushed a commit with the changes you suggested. I'll change the commit message when I squash that in (I'll wait a bit to squash so it's clear what the new changes are).

**Comment by seelabs** [comment]
Squashed and rebased onto b5

**Pull request merged**